FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Common/Configurations/ConfigurationResolvers/ConfigurationResolverV1.cs

```diff
@@ -15,8 +15,19 @@ public class ConfigurationResolverV1
             IServiceCollection services,
             IConfiguration configuration)
         {
-            services.Configure<FhirServerConfiguration>(options =>
-                configuration.GetSection(ConfigurationConstants.FhirServerConfigurationKey).Bind(options));
+            // Try get DataSourceConfiguration first. If not found, create one from FhirServerConfiguration.
+            // If both exist, DataSourceConfiguration takes effect.
+            if (configuration.GetSection(ConfigurationConstants.DataSourceConfigurationKey).Exists())
```

Comment:
Maybe check whether **fhirServer** key exist. If **fhirServer** and **dataSource** both exist, firstly take **fhirServer** section as the data source configuration.

Reply:
By an offline discussion with @moria97, we plan to have higher priority of `dataSource`, it is more general and can be used for other data sources as well. `fhirServer` is also supported for backward compatibility. If customers do not specify the latest configuration with `dataSource`, we will fall back to find `fhirServer`.